From 8cfafbaf7c35713310a8661a4e6fced6b80da80c Mon Sep 17 00:00:00 2001
From: Hongcheng Xie <hongcheng.xie@intel.com>
Date: Wed, 6 Apr 2022 11:46:29 +0800
Subject: [PATCH] Skip do sysrq in android container

Skip doSysRq in android container since it may hang host OS.

Tracked-On: OAM-101737
Signed-off-by: Hongcheng Xie <hongcheng.xie@intel.com>
Change-Id: Ic4a5ebec2ae3081bbac19f01a7184940b22443c7
---
 services/core/java/com/android/server/Watchdog.java | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/services/core/java/com/android/server/Watchdog.java b/services/core/java/com/android/server/Watchdog.java
index 425a0452b6b8..f68dd32aa32a 100644
--- a/services/core/java/com/android/server/Watchdog.java
+++ b/services/core/java/com/android/server/Watchdog.java
@@ -654,8 +654,9 @@ public class Watchdog extends Thread {
             report.append(tracesFileException.getBuffer());
 
             // Trigger the kernel to dump all blocked threads, and backtraces on all CPUs to the kernel log
-            doSysRq('w');
-            doSysRq('l');
+            //doSysRq('w');
+            //doSysRq('l');
+            Slog.w(TAG, "Skip doSysRq in android container since it may hang host OS!");
 
             // Try to add the error to the dropbox, but assuming that the ActivityManager
             // itself may be deadlocked.  (which has happened, causing this statement to
-- 
2.25.1

